src/AutoRest.CSharp/Mgmt/Output/MgmtExtensions.cs

```diff
@@ -116,16 +113,51 @@ private IEnumerable<Segment> GetExtraLayers(RequestPath requestPath, Resource re
         internal static Resource? GetResourceFromResourceType(Operation operation, BuildContext<MgmtOutputLibrary> context)
```

Comment:
Looks like this method does not belong to this class anymore? Maybe we could consider to move it elsewhere